x/stakeibc/keeper/msg_server.go

```diff
@@ -1103,3 +1104,47 @@ func (k msgServer) ResumeHostZone(goCtx context.Context, msg *types.MsgResumeHos
 
 	return &types.MsgResumeHostZoneResponse{}, nil
 }
+
+// Submits an ICA tx to either grant or revoke authz permisssions to an address
+// to execute trades on behalf of the trade ICA
+func (k msgServer) ToggleTradeController(
+	goCtx context.Context,
+	msg *types.MsgToggleTradeController,
+) (*types.MsgToggleTradeControllerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Fetch the trade ICA which will be the granter
+	tradeRoute, found := k.GetTradeRouteFromTradeChainId(ctx, msg.ChainId)
+	if !found {
+		return nil, types.ErrTradeRouteNotFound.Wrapf("trade route not found for chain ID %s", msg.ChainId)
+	}
+
+	// Build the authz message that grants or revokes trade permissions to the specified address
+	authzMsg, err := k.BuildTradeAuthzMsg(ctx, tradeRoute, msg.PermissionChange, msg.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the ICA channel owner from the trade route
+	tradeRouteAccountOwner := types.FormatTradeRouteICAOwnerFromRouteId(
+		msg.ChainId,
+		tradeRoute.GetRouteId(),
+		types.ICAAccountType_CONVERTER_TRADE,
+	)
+
+	// Submit the ICA tx from the trade ICA account
+	// Timeout the ICA at 1 hour
```

Comment:
extremely nit: should we time this out at 1 hour, or on the epoch? Conceptually, this feels completely disjoint from the rest of the stakeibc logic, but I believe we time out most of the other stakeibc ICA messages at the `stride_epoch`

Reply:
I don't think we should time it out at the stride epoch, cause then we have to think about the timing of when we send the message (e.g. if we submit the message at the epoch boundary, we could automatically get a timeout)

imo, we should only use the stride epoch timeout for things that are triggered in our epochly hooks and can't have any overlap across epochs

Reply:
This makes sense to me! Resolving